apollo-router/src/plugins/file_uploads/multipart_form_data.rs

```diff
@@ -0,0 +1,89 @@
+use bytes::Bytes;
+use bytes::BytesMut;
+use futures::stream::StreamExt;
+use futures::stream::TryStreamExt;
+use futures::Stream;
+use http::HeaderMap;
+use http::HeaderValue;
+use mediatype::names::BOUNDARY;
+use mediatype::names::FORM_DATA;
+use mediatype::names::MULTIPART;
+use mediatype::MediaType;
+use rand::RngCore;
+
+use super::MapField;
+use super::MultipartRequest;
+use super::UploadResult;
+
+pub(super) struct MultipartFormData {
+    boundary: String,
+    operations: hyper::Body,
+    map: MapField,
+    multipart: MultipartRequest,
+}
+
+impl MultipartFormData {
+    pub(super) fn new(operations: hyper::Body, map: MapField, multipart: MultipartRequest) -> Self {
+        let boundary = format!(
+            "------------------------{:016x}",
```

Comment:
if a custom boundary is used, it should be called out as part of the `Content-Type` header: https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html#z0

```
Content-Type: multipart/mixed; boundary=gc0p4Jq0M2Yt08jU534c0p
```

Reply:
also, why the `------------------------` prefix? It is not necessary

Reply:
> if a custom boundary is used, it should be called out as part of the `Content-Type` header: https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html#z0
> 
> ```
> Content-Type: multipart/mixed; boundary=gc0p4Jq0M2Yt08jU534c0p
> ```

Correct, I provide it here: https://github.com/apollographql/router/pull/4584/files#diff-250dfae3c4048ea64ae9528e5286bfbbc8af08b7bec7d8bdb279d97385995d1eR43